packages/shopify-app-remix/src/components/RemixPolarisLink.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import {Link, LinkProps as RemixLinkProps} from '@remix-run/react';
+// Polaris' LinkProps doesn't match this type, so we need to import it directly
+import {LinkLikeComponentProps} from '@shopify/polaris/build/ts/src/utilities/link';
+
+export function RemixPolarisLink(props: LinkLikeComponentProps) {
+  const {url, children, onClick, target, ref, ...rest} = props;
+
+  const remixProps: RemixLinkProps = {} as any;
+  if (url) {
+    remixProps.to = url;
+  }
+  if (onClick) {
+    remixProps.onClick = () => {
+      onClick(undefined as any);
+    };
+  }
+  if (target) {
+    remixProps.target = target;
+  }
+
+  return (
+    <Link ref={ref as React.Ref<HTMLAnchorElement>} {...remixProps} {...rest}>
+      {props.children}
+    </Link>
+  );
+}
```

Comment:
Almost none of this should be needed. Also, `props.ref` won't be there, you have to use forwardRef to obtain the outer ref from a function component.

```suggestion
export const RemixPolarisLink = forwardRef<HTMLAnchorElement, LinkLikeComponentProps>((props, ref) => {
  return <Link {...props} to={props.url ?? props.to} ref={ref} />;
})
```

Reply:
I added the explicit checks because I wasn't sure if explicitly setting the `to` property would work (it's a mandatory prop), and the `onClick` handlers for them don't match up, so we need to have the intermediate function. `target` can probably be dropped though.